kahuna/public/js/search/index.js

```diff
@@ -47,6 +49,11 @@ search.config(['$stateProvider',
                 });
                 return {state: $dsr$.redirect.state, params};
             }]
+        },
+        resolve: {
+            searchQueryService: ['$stateParams', 'searchQueryService', function($stateParams, searchQueryService) {
+                return searchQueryService($stateParams.query);
```

Comment:
@theefer the problem I have here is that `$stateParms.query` is formed when we're in the `search.results` view, but we need it in both. I was wondering if the view hierarchy that we have at the moment is necessarily correct.